### ISSUE TITLE
Distant Thunder mission gives lua error

### ISSUE THREAD

**Issue by AMHOL** [OpenRA/OpenRA#15449]
### Issue Summary
Distant thunder mission gives lua error
... ... ...

#### System Information
- Ubuntu 16.04 LTS
- Mono JIT compiler version 5.10.1.47 (tarball Tue Apr 17 15:12:35 UTC 2018)
- playtest-20180729
- Red Alert

#### Additional Information:
- Steps to reproduce
	1. Start distant thunder mission

- Logs
```
Fatal Lua Error: 
Stack Traceback
===============
(2) global C function 'error'
(3) Lua function '(anonymous)' at file '/tmp/.mount_OpenRAr6Wm3X/usr/lib/openra//lua/sandbox.lua:150' (best guess)
  Local variables:
   arg = nil
   ok = boolean: false
   result = string: "[string \"BindingSupport.lua\"]:30: Unable to convert parameter 0 to Actor"
(4) tail call
(5) Lua function '?' at line 23 of chunk '"scriptwrapper.lua"]'
(6) global C function 'xpcall'
(7) Lua upvalue 'TryRunSandboxed' at line 23 of chunk '"scriptwrapper.lua"]'
  Local variables:
   fn = Lua function '?' (defined at line 192 of chunk "soviet05.lua"])
(8) Lua function '?' at line 31 of chunk '"scriptwrapper.lua"]'

Error message
===============
.../.mount_OpenRAr6Wm3X/usr/lib/openra//lua/sandbox.lua:150: [string "BindingSupport.lua"]:30: Unable to convert parameter 0 to Actor
===============
  at OpenRA.Scripting.ScriptContext.FatalError (System.String message) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod , System.Object , System.Object[] , System.Exception& ) [0x00000] in <9bbab8f8a2a246e98480e70b0839fd67>:0 
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <9bbab8f8a2a246e98480e70b0839fd67>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <9bbab8f8a2a246e98480e70b0839fd67>:0 
  at Eluant.LuaRuntime+MethodWrapper.Invoke (System.Object[] parms) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MakeManagedCall (System.IntPtr state, Eluant.LuaRuntime+MethodWrapper wrapper) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MethodWrapperCallCallback (System.IntPtr state) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MethodWrapperCallCallbackWrapper (System.IntPtr state) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaApi.lua_pcall (System.IntPtr , System.Int32 , System.Int32 , System.Int32 ) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.Call (System.Collections.Generic.IList`1[T] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.Call (Eluant.LuaFunction fn, System.Collections.Generic.IList`1[T] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaFunction.Call (System.Collections.Generic.IList`1[T] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaFunction.Call (Eluant.LuaValue[] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at OpenRA.Scripting.ScriptContext.WorldLoaded () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Mods.Common.Scripting.LuaScript.OpenRA.Traits.IWorldLoaded.WorldLoaded (OpenRA.World world, OpenRA.Graphics.WorldRenderer worldRenderer) [0x0002d] in <14a99c21684346b0b1c037d5e702c635>:0 
  at OpenRA.World.LoadComplete (OpenRA.Graphics.WorldRenderer wr) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.StartGame (System.String mapUID, OpenRA.WorldType type) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Network.UnitOrders.ProcessOrder (OpenRA.Network.OrderManager orderManager, OpenRA.World world, System.Int32 clientId, OpenRA.Order order) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Network.OrderManager.TickImmediate () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Sync+<CheckSyncUnchanged>c__AnonStorey0.<>m__0 () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Sync.CheckSyncUnchanged[T] (OpenRA.World world, System.Func`1[TResult] fn) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Sync.CheckSyncUnchanged (OpenRA.World world, System.Action fn) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.LogicTick () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.Loop () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.Run () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
Fatal Lua Error: 
Stack Traceback
===============
(2) global C function 'error'
(3) Lua function '(anonymous)' at file '/tmp/.mount_OpenRAr6Wm3X/usr/lib/openra//lua/sandbox.lua:150' (best guess)
  Local variables:
   arg = nil
   ok = boolean: false
   result = string: "[string \"BindingSupport.lua\"]:30: Unable to convert parameter 0 to Actor"
(4) tail call
(5) Lua function '?' at line 23 of chunk '"scriptwrapper.lua"]'
(6) global C function 'xpcall'
(7) Lua upvalue 'TryRunSandboxed' at line 23 of chunk '"scriptwrapper.lua"]'
  Local variables:
   fn = Lua function '?' (defined at line 192 of chunk "soviet05.lua"])
(8) Lua function '?' at line 31 of chunk '"scriptwrapper.lua"]'

Error message
===============
.../.mount_OpenRAr6Wm3X/usr/lib/openra//lua/sandbox.lua:150: [string "BindingSupport.lua"]:30: Unable to convert parameter 0 to Actor
===============
  at OpenRA.Scripting.ScriptContext.FatalError (System.String message) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod , System.Object , System.Object[] , System.Exception& ) [0x00000] in <9bbab8f8a2a246e98480e70b0839fd67>:0 
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <9bbab8f8a2a246e98480e70b0839fd67>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <9bbab8f8a2a246e98480e70b0839fd67>:0 
  at Eluant.LuaRuntime+MethodWrapper.Invoke (System.Object[] parms) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MakeManagedCall (System.IntPtr state, Eluant.LuaRuntime+MethodWrapper wrapper) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MethodWrapperCallCallback (System.IntPtr state) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MethodWrapperCallCallbackWrapper (System.IntPtr state) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaApi.lua_pcall (System.IntPtr , System.Int32 , System.Int32 , System.Int32 ) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.Call (System.Collections.Generic.IList`1[T] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.Call (Eluant.LuaFunction fn, System.Collections.Generic.IList`1[T] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaFunction.Call (System.Collections.Generic.IList`1[T] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaFunction.Call (Eluant.LuaValue[] args) [0x00000] in <c8bc0b4612b543268b8307855c69585a>:0 
  at OpenRA.Scripting.ScriptContext.WorldLoaded () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Mods.Common.Scripting.LuaScript.OpenRA.Traits.IWorldLoaded.WorldLoaded (OpenRA.World world, OpenRA.Graphics.WorldRenderer worldRenderer) [0x0002d] in <14a99c21684346b0b1c037d5e702c635>:0 
  at OpenRA.World.LoadComplete (OpenRA.Graphics.WorldRenderer wr) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.StartGame (System.String mapUID, OpenRA.WorldType type) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Network.UnitOrders.ProcessOrder (OpenRA.Network.OrderManager orderManager, OpenRA.World world, System.Int32 clientId, OpenRA.Order order) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Network.OrderManager.TickImmediate () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Sync+<CheckSyncUnchanged>c__AnonStorey0.<>m__0 () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Sync.CheckSyncUnchanged[T] (OpenRA.World world, System.Func`1[TResult] fn) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Sync.CheckSyncUnchanged (OpenRA.World world, System.Action fn) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.LogicTick () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.Loop () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.Run () [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00000] in <01d2db41a5cd4227954eb0c882b05315>:0 
```

- Screenshots & Videos
![selection_032](https://user-images.githubusercontent.com/1582784/43600517-5c113ec0-96bd-11e8-87c7-3baf3ab544c5.png)

**Comment by FrameLimiter** [comment]
Also the case for Allies: 4a, 4b, and Soviet 4a, 4b.

**Comment by abcdefg30** [comment]
The d2k crashes will be fixed by #14795.
